app/util/ClassToMap.scala

```diff
@@ -0,0 +1,101 @@
+package util
+
+import com.gu.contentatom.thrift.atom.media._
+import com.gu.contentatom.thrift.{ImageAssetDimensions, ImageAsset, Image}
+import shapeless.labelled._
+import shapeless._
+
+object ClassToMap {
+  // FROM CASE CLASS TO MAP (http://stackoverflow.com/a/31638390)
+  trait ToMapRec[L <: HList] {
+    def apply(l: L): Map[String, Any]
+  }
+
+  // Case for heads that don't have ToMapRec instances (hconsToMapRec1).
+  // Note that we need to use a LowPriority trait to make sure that this instance is
+  // prioritized properly with respect to hconsToMapRec0—if we didn't, the two would have
+  // the same priority and we'd get errors about ambiguous instances.
+  trait LowPriorityToMapRec {
+    implicit def hconsToMapRecNode[K <: Symbol, V, T <: HList]
+    (implicit
+     wit: Witness.Aux[K],
+     tmrT: Lazy[ToMapRec[T]]
```

Comment:
It would maybe be nice to give `tmrT` and `tmrH` different names as it makes it a bit hard to read as I have to keep remembering what they are in quite a complex bit of code